oci/layout/oci_transport.go

```diff
@@ -228,7 +247,55 @@ func (ref ociReference) NewImageDestination(ctx context.Context, sys *types.Syst
 
 // DeleteImage deletes the named image from the registry, if supported.
 func (ref ociReference) DeleteImage(ctx context.Context, sys *types.SystemContext) error {
-	return errors.New("Deleting images not implemented for oci: images")
+	// Get the manifest for the image
+	descriptor, err := ref.getManifestDescriptor()
+	if err != nil {
+		return err
+	}
+
+	manifest, err := ref.getManifest(descriptor)
+	if err != nil {
+		return err
+	}
+
+	// Delete all blobs
+	for _, d := range append(manifest.Layers, manifest.Config, descriptor) {
+		blobPath, err := ref.blobPath(d.Digest, "")
```

Comment:
Blobs can be shared across images; deleting one image should not break the others.

Reply:
Yes indeed, I'm working on the test setup for this case.